int32 agentId
int32 startX
int32 startY

